### PR TITLE
Allow imagestreamtag delete to retry conflicts

### PR DESCRIPTION
**Pull request by smarterclayton**
Also return a structured error instead of an opaque error so that
clients can react when unknown problems occur.

@bparees dual to the other one

### CODE REVIEW

File: pkg/image/registry/imagestreamtag/rest.go [hunk 1]
```diff
@@ -315,34 +315,47 @@ func (r *REST) Delete(ctx apirequest.Context, id string, options *metav1.DeleteO
 		return nil, false, err
 	}
 
-	stream, err := r.imageStreamRegistry.GetImageStream(ctx, name, &metav1.GetOptions{})
-	if err != nil {
-		return nil, false, err
-	}
+	for i := 10; i > 0; i-- {
```
Comment (miminar) [inline pkg/image/registry/imagestreamtag/rest.go]: Out of curiosity: why exponential backoff isn't used here?

File: pkg/image/registry/imagestreamtag/rest.go [hunk 2]
```diff
@@ -315,34 +315,47 @@ func (r *REST) Delete(ctx apirequest.Context, id string, options *metav1.DeleteO
...
+		notFound := true
 
-	// Try to delete the status tag
-	if _, ok := stream.Status.Tags[tag]; ok {
-		delete(stream.Status.Tags, tag)
-		notFound = false
-	}
+		// Try to delete the status tag
+		if _, ok := stream.Status.Tags[tag]; ok {
+			delete(stream.Status.Tags, tag)
```
Comment (miminar) [inline pkg/image/registry/imagestreamtag/rest.go]: It looks like ` // The tag history remains intact and is not deleted.` in the godoc does not apply anymore.

File: pkg/image/registry/imagestreamtag/rest.go (hunk 1 above)
Comment (bparees) [reply pkg/image/registry/imagestreamtag/rest.go]: it's a conflict, the assumption is we should be able to fetch the new/updated object and try again immediately.  backing off wouldn't provide any value in that scenario unless we think there is a storm of updates occurring, in which case we'd want a random backoff, not an exponential one.

**Comment by smarterclayton** [comment]
Updated comment and fixed test error.   Reapplying label.

**Pull request merged**